tests/artifacts_tests/test_list_artifact_meta.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import pathlib
+
+import pytest
+
+import optuna
+from optuna.artifacts import FileSystemArtifactStore
+from optuna.artifacts import get_all_artifact_meta
+from optuna.artifacts import upload_artifact
+from optuna.artifacts._protocol import ArtifactStore
+from optuna.storages import BaseStorage
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import Trial
+
+
+@pytest.fixture(params=["FileSystem"])
+def artifact_store(tmp_path: pathlib.PurePath, request: pytest.FixtureRequest) -> ArtifactStore:
+    if request.param == "FileSystem":
+        return FileSystemArtifactStore(str(tmp_path))
+    assert False, f"Unknown artifact store: {request.param}"
+
+
+def _check_artifact_meta(
+    artifact_store: ArtifactStore,
+    study_or_trial: Trial | FrozenTrial | Study,
+    file_path: str,
+    filename: str,
+    mimetype: str | None,
+    encoding: str | None,
+    storage: BaseStorage | None,
+    n_linked_artifact_meta: int,
+) -> None:
+    artifact_id = upload_artifact(
+        study_or_trial,
+        file_path,
+        artifact_store,
+        storage=storage,
+        mimetype=mimetype,
+        encoding=encoding,
+    )
+    mimetype = "text/plain" if mimetype is None else mimetype
+    if isinstance(study_or_trial, FrozenTrial):
+        artifact_meta_list = get_all_artifact_meta(study_or_trial, storage=storage)
+    else:
+        artifact_meta_list = get_all_artifact_meta(study_or_trial)
```

Comment:
It looks like a kind of conditional test logic which is described at xUnit Test Patterns.
http://xunitpatterns.com/Conditional%20Test%20Logic.html

Is it possible to simplify this logic?

Reply:
Thank you for the suggestion!

Done!